forte/data/readers/classification_reader.py

```diff
@@ -0,0 +1,177 @@
+# Copyright 2020 The Forte Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import csv
+import importlib
+from typing import Iterator, Tuple
+from collections import OrderedDict
+from forte.common import Resources
+from forte.common.configuration import Config
+from forte.data.data_pack import DataPack
+from forte.data.base_reader import PackReader
+from ft.onto.base_ontology import *
+
+
+__all__ = ["ClassificationDatasetReader"]
+
+
+class ClassificationDatasetReader(PackReader):
+    r"""
+    A generic dataset reader class for classification datasets.
+
+    """
+
+    def __init__(self):
+        super().__init__()
+        self.index2class = None
+        self.label2index = None
+
+    def set_up(self):
+        assert self.configs.index2class is not None, (
+            "User must set index2class to enable"
+            " the dataset reader encode labels correctly"
+        )
+
+        # class and index
+        if not self.configs.digit_label:
+            # initialize class
+            self.index2class = self.configs.index2class
+            self.label2index = {
+                v: k for k, v in self.index2class.items()
+            }
+
+    def initialize(self, resources: Resources, configs: Config):
+        super().initialize(resources, configs)
+        self.set_up()
+
+    def _collect(self, csv_file: str) -> Iterator[Tuple[int, str]]:
+        with open(csv_file, encoding="utf-8") as f:
+            data = csv.reader(f, delimiter=",", quoting=csv.QUOTE_ALL)
+            next(data)
+            for line_id, line in enumerate(data):
+                yield line_id, line
+
+    def _cache_key_function(self, line_info: Tuple[int, str]) -> str:
+        return str(line_info[0])
+
+    def _parse_pack(self, line_info: Tuple[int, str]) -> Iterator[DataPack]:
```

Comment:
The type annotation should be wrong, it should be something like `Tuple[int, List[str]]` (depends on what CSV reader would output).